app/src/main/java/app/revanced/manager/ui/component/AlertDialogExtended.kt

```diff
@@ -0,0 +1,153 @@
+                                Modifier
+                                    .padding(bottom = 16.dp)
+                                    .align(
+                                        if (icon == null) {
+                                            Alignment.Start
+                                        } else {
+                                            Alignment.CenterHorizontally
+                                        }
+                                    )
+                            ) {
+                                title()
+                            }
+                        }
+                    }
+                }
+                text?.let {
+                    ContentStyle(
+                        color = textContentColor,
+                        textStyle = MaterialTheme.typography.bodyMedium
+                    ) {
+                        Box(
+                            Modifier
+                                .weight(weight = 1f, fill = false)
+                                .padding(bottom = 24.dp)
+                                .padding(textHorizontalPadding)
+                                .align(Alignment.Start)
+                        ) {
+                            text()
+                        }
+                    }
+                }
+                Box(
+                    modifier = Modifier
+                        .padding(horizontal = 24.dp)
+                ) {
+                    ContentStyle(
+                        color = MaterialTheme.colorScheme.primary,
+                        textStyle = MaterialTheme.typography.labelLarge
+                    ) {
+                        FlowRow(
+                            modifier = Modifier.fillMaxWidth(),
+                            horizontalArrangement = Arrangement.spacedBy(
+                                12.dp,
+                                if (tertiaryButton != null) Alignment.Start else Alignment.End
+                            ),
+                            verticalArrangement = Arrangement.spacedBy(8.dp)
+                        ) {
+                            tertiaryButton?.let {
+                                it()
+                                Spacer(modifier = Modifier.weight(1f))
+                            }
+                            dismissButton?.invoke()
+                            confirmButton()
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun ContentStyle(
+    color: Color? = null,
+    textStyle: TextStyle? = null,
+    content: @Composable () -> Unit
+) {
+    val contentColor = color ?: LocalContentColor.current
+    val finalTextStyle = textStyle ?: LocalTextStyle.current
+
+    CompositionLocalProvider(LocalContentColor provides contentColor) {
+        ProvideTextStyle(finalTextStyle) {
+            content()
+        }
+    }
+}
```

Comment:
I think I have seen compose-m3 code do something like this:

```suggestion
@Composable
private fun ContentStyle(
    color: Color = LocalContentColor.current,
    textStyle: TextStyle = LocalTextStyle.current,
    content: @Composable () -> Unit
) {
    CompositionLocalProvider(LocalContentColor provides color) {
        ProvideTextStyle(textStyle) {
            content()
        }
    }
}
```

Make sure the code still compiles though.